BuddySave/Core/GamingSession.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Diagnostics;
+using BuddySave.Core.Models;
+using BuddySave.System;
+using NLog;
+
+namespace BuddySave.Core;
+
+public class GamingSession : IGamingSession
+{
+    private readonly ILogger _logger;
+    private readonly ISharedSaveOrchestrator _sharedSaveOrchestrator;
+    private readonly IProcessProvider _processProvider;
+
+    public GamingSession(ILogger logger, ISharedSaveOrchestrator sharedSaveOrchestrator, IProcessProvider processProvider)
+    {
+        _logger = logger;
+        _sharedSaveOrchestrator = sharedSaveOrchestrator;
+        _processProvider = processProvider;
+    }
+
+    public async Task Run(GameSave gameSave, Session session, ServerParameters serverParameters)
+    {
+        if (string.IsNullOrWhiteSpace(serverParameters.Path))
+        {
+            throw new ArgumentException("No server path provided. Cannot start a gaming session.");
+        }
+
+        await _sharedSaveOrchestrator.Load(gameSave, session);
+        await RunServer(serverParameters);
+        await _sharedSaveOrchestrator.Save(gameSave, session);
+    }
+
+    private async Task RunServer(ServerParameters serverParameters)
+    {
+        var workingDirectory = Path.GetDirectoryName(serverParameters.Path);
+        var startInfo = new ProcessStartInfo()
+        {
+            FileName = serverParameters.Path,
+            Arguments = serverParameters.Arguments,
+            WorkingDirectory = workingDirectory,
+            UseShellExecute = true
+        };
+
+        var process = _processProvider.Start(startInfo);
+        var serverString = string.IsNullOrEmpty(serverParameters.Arguments)
+            ? serverParameters.Path
+            : $"{serverParameters.Path} {serverParameters.Arguments}";
+        _logger.Info(@$"Server started, waiting for exit: ""{serverString}""");
+
+        await _processProvider.WaitForExitAsync(process);
```

Comment:
How does the process know when to exit in our case? When we close the console window by hand?

Reply:
Yes. In essence this is the change. From:
* Launch Buddy Save
* Load
* Launch Server
* Close Server
* Save

It becomes:
* Launch Buddy Save
* Run (load + launch server)
* Close server (+ save)

Reply:
If there's an auto-save, it would be worth to have a method or at least a comment somewhere indicating this functionality :)

Reply:
Renamed some methods for clarity.